management/commands/sync_users_with_nodebb.py

```diff
@@ -0,0 +1,49 @@
+"""
+Django management command to create users at NodeBB corresponding to edX users.
+"""
+from logging import getLogger
+
+from django.contrib.auth.models import User
+from django.core.management.base import BaseCommand
+from openedx.features.openedx_nodebb_discussion.client.tasks import (
+    task_create_user_on_nodebb, task_sync_user_profile_info_with_nodebb
+)
+from openedx.features.openedx_nodebb_discussion.models import EdxNodeBBUser
+from student.models import UserProfile
+
+log = getLogger(__name__)
+
+
+class Command(BaseCommand):
+    help = """
+    This command creates users in NodeBB according to all User and UserProfile instances in edX.
+
+    Example usage:
+        manage.py ... sync_users_with_nodebb
+    """
+
+    def handle(self, *args, **options):
+        nodebb_relations = EdxNodeBBUser.objects.all()
+        edx_users = User.objects.all()
+        user_profiles = UserProfile.objects.all()
+
+        for edx_user in edx_users:
+            nodebb_user = nodebb_relations.filter(edx_uid=edx_user)
+            if not nodebb_user:
+                user_data = {
+                    'username': edx_user.username,
+                    'email': edx_user.email,
+                    'joindate': edx_user.date_joined.strftime("%s")
+                }
+                task_create_user_on_nodebb.delay(**user_data)
+
+                profile = user_profiles.filter(user=edx_user)
```

Comment:
can we use get_object_or_404 ?

Reply:
Using `get_object_or_404 ` we will get 404 exception if the record is not found so we will have to handle that through `try catch` block which we don't want to do here. so using `filter` is best for here.

Reply:
done in the lastest commit.